lib/X25519KeyAgreementKey2020.js

```diff
@@ -372,69 +356,45 @@ X25519KeyAgreementKey2020.SUITE_CONTEXT =
+    return false;
+  }
 
-/**
- * Decodes a given multibase-encoded string.
- *
- * @param {string} text - Multibase encoded string to decode.
- * @returns {Uint8Array} Decoded bytes.
- */
-function _multibaseDecode({text}) {
-  // drop the initial multibase 'z' prefix
-  return base58btc.decode(text.substr(1));
+  const keyBytes = base58btc.decode(multibaseKey.slice(1));
+  return expectedHeader.every((val, i) => keyBytes[i] === val);
 }
 
 /**
- * Returns raw fingerprint bytes, by adding a multicodec prefix for an
- * X25519 public key.
- *
- * @see https://github.com/multiformats/multicodec/blob/master/table.csv
- * 0xec is the value for X25519 public key
- * 0x01 is from varint.encode(0xec) -> [0xec, 0x01]
- * @see https://github.com/multiformats/unsigned-varint
- *
- * @param {object} [options={}] - Options hashmap.
- * @param {Uint8Array} options.publicKeyBytes - Public key raw bytes.
+ * Encodes a given Uint8Array to multibase-encoded string.
  *
- * @returns {Uint8Array} Multicodec formatted public key fingerprint bytes.
+ * @param {Uint8Array} header - Multicodec header to prepend to the bytes.
+ * @param {Uint8Array} bytes - Bytes to encode.
+ * @returns {string} Multibase-encoded string.
  */
-function _multicodecFormatFingerprint({publicKeyBytes} = {}) {
-  const fingerprintBytes = new Uint8Array(2 + publicKeyBytes.length);
-  fingerprintBytes[0] = 0xec;
-  fingerprintBytes[1] = 0x01;
-  fingerprintBytes.set(publicKeyBytes, 2);
-  return fingerprintBytes;
+function _multibaseEncode(header, bytes) {
+  const mcBytes = new Uint8Array(header.length + bytes.length);
+
+  mcBytes.set(header);
+  mcBytes.set(bytes, header.length);
+
+  return MULTIBASE_BASE58BTC_HEADER + base58btc.encode(mcBytes);
 }
 
 /**
- * Decodes a multicodec-formatted Uint8Array containing an x25519 public key.
- *
- * @see https://github.com/multiformats/multicodec/blob/master/table.csv
- * 0xec is the value for X25519 public key
- * 0x01 is from varint.encode(0xec) -> [0xec, 0x01]
- * @see https://github.com/multiformats/unsigned-varint
- *
- * @param {object} [options={}] - Options hashmap.
- * @param {Uint8Array} options.bytes - Key fingerprint decoded from multibase.
+ * Decodes a given multibase-encoded string.
  *
- * @returns {{publicKeyBytes: Uint8Array}} The public key bytes (without the
- *   multicodec prefix).
+ * @param {string} text - Multibase encoded string to decode.
+ * @returns {Uint8Array} Decoded bytes.
  */
-function _multicodecDecodeFingerprint({bytes} = {}) {
-  if(!(bytes && bytes[0] === 0xec && bytes[1] === 0x01)) {
-    // eslint-disable-next-line max-len
-    throw new Error('Expecting public key to be "x255519-pub" multicodec formatted [0xec, 0x01, <public key bytes>]');
-  }
-  // Remove the multicodec prefix
-  const publicKeyBytes = bytes.slice(2);
-  return publicKeyBytes;
+function _multibaseDecode(text) {
+  const mcValue = base58btc.decode(text.substr(1));
+  return mcValue.slice(2);
```

Comment:
Did we want to check the key header here? The above check got removed and isn't performed elsewhere.

Reply:
Fixed in 5f79c9c90b7aa35f1d4ea5c889506b0b26c9961e.